augur/export_v2.py

```diff
@@ -846,6 +831,49 @@ def set_description(data_json, cmd_line_description_file):
     except FileNotFoundError:
         fatal("Provided desciption file {} does not exist".format(cmd_line_description_file))
 
+def transfer_mutations_to_branches(node_attrs, branch_attrs):
+    for node_name, raw_data in node_attrs.items():
+        if "aa_muts" in raw_data or "muts" in raw_data:
+            if node_name not in branch_attrs:
+                branch_attrs[node_name] = {}
+            branch_attrs[node_name]["mutations"] = {}
+            if "muts" in raw_data and len(raw_data["muts"]):
+                branch_attrs[node_name]["mutations"]["nuc"] = raw_data["muts"]
+            if "aa_muts" in raw_data:
+                aa = {gene:data for gene, data in raw_data["aa_muts"].items() if len(data)}
+                branch_attrs[node_name]["mutations"].update(aa)
+                #convert mutations into a label
+                if aa:
+                    aa_lab = '; '.join("{!s}: {!s}".format(key,', '.join(val)) for (key,val) in aa.items())
+                    if 'labels' in branch_attrs[node_name]:
+                        branch_attrs[node_name]["labels"]["aa"] = aa_lab
+                    else:
+                        branch_attrs[node_name]["labels"] = { "aa": aa_lab }
+
+def transfer_clade_annotation_to_branches(node_attrs, branch_attrs):
+    # NOTE: storing branch labels as `clade_annotation` is only possible
+    # using an older version of augur. (`augur clades` no longer uses this).
+    # This function should be removed upon the next major augur release (currently v12)
```

Comment:
I'm pretty sure I won't remember to remove this function on the next major Augur release. 😆 How do you feel about removing this function now and releasing this PR in a major release?

If users have updated Augur to the version with this PR already, then `augur export` will know how to handle the new `branch_labels` output from `augur clades`, right? So, this change away from the `clade_annotation` attribute would only be an issue for users who have generated clade JSONs with an older version of Augur and then tried to export them with the new version. Or am I missing something else?

 In the worst case scenario, users would have to re-run `augur clades` on their data to get updated JSONs, right?

Reply:
Done! This PR will now necessitate a major augur release. 

> In the worst case scenario, users would have to re-run augur clades on their data to get updated JSONs, right?

Yes, and `augur export` will now detect the `generated_by` mismatch and require this. `augur export v2` will issue a warning that they need to rename a key in their auspice config file, but will still export clade as it's coming from a node-data file